### PR TITLE
matrix genes menu: at editing a group, move the group selector from s…

### LINKED ISSUE #1780: matrix genes menu: at editing a group, move the group selector from secondary menu to primary menu
updated [ui sketch](https://docs.google.com/drawings/d/1zynSoWdkLLGGrZ79uoyXW6meW81Zt1p5-vJcTMZk8y4/edit) at top right

### PR DESCRIPTION
**Pull request by congyu-lu**
…econdary menu to primary menu

## Description
fix #1780 
also fix the "Create Group" error

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/plots/matrix.controls.js [hunk 1]
```diff
@@ -383,20 +383,6 @@ export class MatrixControls {
 					// 	settingsKey: 'geneVariantCountSamplesSkipMclass',
 					// 	processInput: tw => {},
 					// },
-					{
```
Comment (siosonel) [inline client/plots/matrix.controls.js:386]: @congyu-lu these inputs are not moved elsewhere? I think they should be, maybe in the layout. In some cases, the labels are really long and affects the offsets too much

Comment (congyu-lu) [reply client/plots/matrix.controls.js:386]: They are also in the _Variables_

Comment (congyu-lu) [reply client/plots/matrix.controls.js:386]: the two in Variables are the same as the two we deleted here: _termGrpLabelMaxChars_ and _rowlabelmaxchars_, _collabelmaxchars_ and _sampleGrpLabelMaxChars_ are inside of Samples

**Comment by siosonel** [comment]
unrelated issue, but noticed while testing this PR: https://github.com/stjude/proteinpaint/issues/1985

**Comment by siosonel** [comment]
Should the `Top mutated genes` option be shown here when creating a new group, similar to how it's visible when editing a group?

Screenshot 2024-08-06 at 2 01 00 PM

**Comment by xzhou82** [comment]
thanks a lot. please fix these
1. in hiercluster, it's possible the clustering group is not about genes. e.g. metabolite intensity clustering. in such case, the Gene Set ui should not show the Edit Current Group option
2. for the input box after Create Group button, allow to hit enter and submit

**Comment by congyu-lu** [comment]
> thanks a lot. please fix these
> 
> 1. in hiercluster, it's possible the clustering group is not about genes. e.g. metabolite intensity clustering. in such case, the Gene Set ui should not show the Edit Current Group option
> 2. for the input box after Create Group button, allow to hit enter and submit
> 3. per Edgar's comment above, when creating a new group, optionally set mode property as argument to genesetedit: if termdbconfig.queries.snvindel is present, set mode=geneVariant, otherwise do not set mode

Thanks just fixed these issues

**Review by xzhou82 — APPROVED** [review]
thanks a lot

**Pull request merged**